packages/trader/src/AppV2/Components/TradeParameters/GrowthRate/growth-rate.tsx

```diff
@@ -1,26 +1,109 @@
 import React from 'react';
 import clsx from 'clsx';
 import { observer } from 'mobx-react';
-import { TextField } from '@deriv-com/quill-ui';
+import { ActionSheet, TextField } from '@deriv-com/quill-ui';
+import { Skeleton } from '@deriv/components';
 import { Localize } from '@deriv/translations';
 import { useTraderStore } from 'Stores/useTraderStores';
-import { getGrowthRatePercentage } from '@deriv/shared';
+import { CONTRACT_TYPES, getGrowthRatePercentage, isEmptyObject } from '@deriv/shared';
+import Carousel from 'AppV2/Components/Carousel';
+import CarouselHeader from 'AppV2/Components/Carousel/carousel-header';
+import TradeParamDefinition from 'AppV2/Components/TradeParamDefinition';
+import GrowthRatePicker from './growth-rate-picker';
 
 type TGrowthRateProps = {
     is_minimized?: boolean;
 };
 
 const GrowthRate = observer(({ is_minimized }: TGrowthRateProps) => {
-    const { growth_rate, has_open_accu_contract } = useTraderStore();
+    const {
+        accumulator_range_list,
+        growth_rate,
+        is_purchase_enabled,
+        is_trade_enabled,
+        has_open_accu_contract,
+        maximum_ticks,
+        onChange,
+        proposal_info,
+        tick_size_barrier_percentage,
+    } = useTraderStore();
+
+    const [is_open, setIsOpen] = React.useState(false);
+    const is_small_screen = window.innerHeight <= 640;
```

Comment:
Do we have to specific about 640px?
can we use the `useDevice` hook here and check if it is mobile?

Reply:
@balakrishna-deriv is mobile would be applicable if it was about device width but it's about the screen height.
it's a design requirement to have different hight of action sheet based on the screen height:
<img width="1147" alt="Screenshot 2024-08-08 at 10 08 34 AM" src="https://github.com/user-attachments/assets/c9ecbd84-5a17-4f50-8c40-e512c2fdbbc4">